Illegal character in path: Internal storage URI fails when filenames contain square brackets []

### Describe the issue

An IllegalArgumentException (Illegal character in path) error is thrown when Kestra attempts to process a file in internal storage that contains square brackets `[ `and `]` in the filename.

It appears that the internal storage path is being parsed as a strict URI `java.net.URI`, but the filename segments are not being URL-encoded before parsing. This causes the URI creation to fail because brackets are reserved characters in IPv6 literals and are not allowed in the path segment without escaping.

### Error Log

```
2025-11-17T05:47:25.601158Z TRACE java.lang.IllegalArgumentException: Illegal character in path at index 122: kestra:///xxx/xxx-cm-dd-001-worker/executions/2HUc1J8hphtMj1EqMcjqlL/tasks/mogambo-downloadfile/5jLeZr6Quyz8jxlxptG8Nm/[xxx]-27SS_xxx_2nd--4.png.Viewable.png
	at java.base/java.net.URI.create(Unknown Source)
	at io.kestra.storage.minio.MinioStorage.put(MinioStorage.java:295)
	at io.kestra.storage.minio.MinioStorage.put(MinioStorage.java:270)
	at io.kestra.core.storages.StorageInterface.put(StorageInterface.java:188)
	at io.kestra.ee.tenant.PerTenantStorage.put(PerTenantStorage.java:102)
	at io.kestra.core.storages.InternalStorage.putFile(InternalStorage.java:162)
	at io.kestra.core.storages.InternalStorage.putFile(InternalStorage.java:181)
	at io.kestra.plugin.core.http.Download.run(Download.java:135)
	at io.kestra.plugin.core.http.Download.run(Download.java:29)
	at io.kestra.worker.WorkerTaskCallable.lambda$doCall$1(WorkerTaskCallable.java:79)
	at dev.failsafe.Functions.lambda$toCtxSupplier$9(Functions.java:228)
	at dev.failsafe.Functions.lambda$get$0(Functions.java:46)
	at dev.failsafe.internal.TimeoutExecutor.lambda$apply$1(TimeoutExecutor.java:93)
	at dev.failsafe.SyncExecutionImpl.executeSync(SyncExecutionImpl.java:187)
	at dev.failsafe.FailsafeExecutor.call(FailsafeExecutor.java:376)
	at dev.failsafe.FailsafeExecutor.run(FailsafeExecutor.java:220)
	at io.kestra.worker.WorkerTaskCallable.doCall(WorkerTaskCallable.java:79)
	at io.kestra.worker.AbstractWorkerCallable.call(AbstractWorkerCallable.java:64)
	at io.kestra.worker.WorkerSecurityService.callInSecurityContext(WorkerSecurityService.java:10)
	at io.kestra.worker.DefaultWorker.lambda$callJob$19(DefaultWorker.java:972)
	at io.kestra.core.trace.NoopTracer.inCurrentContext(NoopTracer.java:15)
	at io.kestra.worker.DefaultWorker.callJob(DefaultWorker.java:968)
	at io.kestra.worker.DefaultWorker.runAttempt(DefaultWorker.java:926)
	at io.kestra.worker.DefaultWorker.run(DefaultWorker.java:728)
	at io.kestra.worker.DefaultWorker.handleTask(DefaultWorker.java:379)
	at io.kestra.worker.DefaultWorker.lambda$run$8(DefaultWorker.java:294)
	at io.micrometer.core.instrument.internal.TimedRunnable.run(TimedRunnable.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.net.URISyntaxException: Illegal character in path at index 122: kestra:///xxx/xxx-cm-dd-001-worker/executions/2HUc1J8hphtMj1EqMcjqlL/tasks/mogambo-downloadfile/5jLeZr6Quyz8jxlxptG8Nm/[xxx]-27SS_xxx_2nd--4.png.Viewable.png
Expected Behavior Kestra should handle filenames with special characters (like brackets, spaces, etc.) by properly URL-encoding the path segments before creating the URI object. The system should treat [xxx].png as a valid filename.
```


Additional notes: 
Its related to : [GitHub issue 124](https://github.com/kestra-io/storage-minio/issues/124)



### Environment

- Kestra Version:  v1.1.5

--- a/core/src/main/java/io/kestra/plugin/core/http/Download.java
+++ b/core/src/main/java/io/kestra/plugin/core/http/Download.java
@@ -20,6 +20,8 @@ import java.io.BufferedOutputStream;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.net.URI;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicReference;
@@ -123,7 +125,11 @@ public class Download extends AbstractHttp implements RunnableTask<Download.Outp
                     String contentDisposition = response.getHeaders().firstValue("Content-Disposition").orElseThrow();
                     rFilename = filenameFromHeader(runContext, contentDisposition);
                     if (rFilename != null) {
+                        URLEncoder.encode(rFilename, StandardCharsets.UTF_8);
                         rFilename = rFilename.replace(' ', '+');
+                        // brackets are IPv6 reserved characters
+                        rFilename = rFilename.replace("[", "%5B");
+                        rFilename = rFilename.replace("]", "%5D");
                     }
                 }
             }

--- a/core/src/test/java/io/kestra/plugin/core/http/DownloadTest.java
+++ b/core/src/test/java/io/kestra/plugin/core/http/DownloadTest.java
@@ -252,8 +252,25 @@ class DownloadTest {
 
         Download.Output output = task.run(runContext);
 
-        assertThat(output.getUri().toString()).doesNotContain("/secure-path/");
         assertThat(output.getUri().toString()).endsWith("file.with+spaces.txt");
+    }
+
+    @Test
+    void contentDispositionWithBrackets() throws Exception {
+        EmbeddedServer embeddedServer = applicationContext.getBean(EmbeddedServer.class);
+        embeddedServer.start();
+
+        Download task = Download.builder()
+            .id(DownloadTest.class.getSimpleName())
+            .type(DownloadTest.class.getName())
+            .uri(Property.ofValue(embeddedServer.getURI() + "/content-disposition-with-brackets"))
+            .build();
+
+        RunContext runContext = TestsUtils.mockRunContext(this.runContextFactory, task, ImmutableMap.of());
+
+        Download.Output output = task.run(runContext);
+
+        assertThat(output.getUri().toString()).endsWith("file.with%5B%5Dbrackets.txt");
     }
 
     @Controller()
@@ -301,6 +318,11 @@ class DownloadTest {
         public HttpResponse<byte[]> contentDispositionWithSpaceAfterDot() {
             return HttpResponse.ok("Hello World".getBytes())
                 .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"file.with spaces.txt\"");
+        }
+        @Get("content-disposition-with-brackets")
+        public HttpResponse<byte[]> contentDispositionWithBrackets() {
+            return HttpResponse.ok("Hello World".getBytes())
+                .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"file.with[]brackets.txt\"");
         }
     }
 }

